File: lib/features/notificaciones/models/create_notificacion_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'notificacion.dart';

part 'create_notificacion_request.g.dart';

@JsonSerializable()
class CreateNotificacionRequest {
  final String titulo;
  final String mensaje;
  final SegmentoNotificacion segmento;
  final String? edificio;

  CreateNotificacionRequest({
    required this.titulo,
    required this.mensaje,
    required this.segmento,
    this.edificio,
  });

  factory CreateNotificacionRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateNotificacionRequestFromJson(json);
  Map<String, dynamic> toJson() => _$CreateNotificacionRequestToJson(this);
}

File: backend/src/main/java/com/condos/notificacion/dto/CreateNotificacionRequest.java
package com.condos.notificacion.dto;

import com.condos.notificacion.model.SegmentoNotificacion;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class CreateNotificacionRequest {

    @NotBlank
    private String titulo;

    @NotBlank
    private String mensaje;

    @NotNull
    private SegmentoNotificacion segmento;

    private String edificio;
}
